axelor-base/src/main/java/com/axelor/apps/base/service/InternationalServiceImpl.java

```diff
@@ -44,8 +44,8 @@ public String translate(String source, String sourceLanguage, String targetLangu
   public Map<String, String> getProductDescriptionAndNameTranslation(
       Product product, Partner partner, String userLanguage) {
     Map<String, String> translation = new HashMap<>();
-    if (product != null && partner != null && partner.getLanguage() != null) {
-      String partnerLanguage = partner.getLanguage().getCode();
+    if (product != null && partner != null && partner.getLocalization() != null) {
+      String partnerLanguage = partner.getLocalization().getCode();
```

Comment:
no here we need to keep the Language.getCode because we want to know which translation to take into account
So you need to check if localization is not null AND if language is not null